lib/core/connect.js

```diff
@@ -99,6 +99,8 @@ function buildConnector ({ maxCachedSessions, socketPath, timeout, ...opts }) {
         servername,
         session,
         localAddress,
+        // TODO(HTTP/2): Add support for h2c
+        ALPNProtocols: allowH2 ? ['http/1.1', 'h2'] : ['http/1.1'],
```

Comment:
Whenever the flag is passed, the `h2` is advertised as supported, though the  `h1` has higher priority.

cc: @mcollina